homeassistant/components/ecovacs/vacuum.py

```diff
@@ -377,3 +405,59 @@ async def async_send_command(
             await self._device.execute_command(
                 self._capability.custom.set(command, params)
             )
+
+    async def async_get_positions(
+        self,
+    ) -> list[Position]:
+        """Get bot and chargers positions."""
+        _LOGGER.debug("async_get_positions")
+
+        if (
+            not self._capability.map
+            or not self._capability.map.position
+            or not self._capability.map.position.get
+        ):
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="vacuum_get_positions_not_supported",
+            )
+
+        positions = []
+        result = await self._device.execute_command(
+            self._capability.map.position.get[0]
+        )
+        if isinstance(result, dict) and result.get("ret") == "ok":
+            data = result.get("resp", {}).get("body", {}).get("data", {})
+
+            for type_str in ("deebotPos", "chargePos"):
+                data_positions = data.get(type_str, [])
+
+                if isinstance(data_positions, dict):
+                    positions.append(
+                        Position(
+                            type=PositionType(type_str),
+                            x=data_positions["x"],
+                            y=data_positions["y"],
+                            a=data_positions.get("a", 0),
+                        )
+                    )
+                else:
+                    positions.extend(
+                        [
+                            Position(
+                                type=PositionType(type_str),
+                                x=entry["x"],
+                                y=entry["y"],
+                                a=entry.get("a", 0),
+                            )
+                            for entry in data_positions
+                        ]
+                    )
+
+        if not positions:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="vacuum_get_positions_invalid",
+            )
+
+        return positions
```

Comment:
```suggestion
        return = await self._device.execute_command(
            self._capability.map.position.get[0]
        )
```

Command response parsing should not be done in HA but in the library.
However, as your use case is very rare and returning also the parsed response, requires a big refactoring in the library. I would suggest that we return the raw response for now.

We can always add later a parsed service, if the use case is bigger. For that reason I added `raw`